src/SLIP10Node.ts

```diff
@@ -64,6 +68,21 @@ export type JsonSLIP10Node = {
    * The name of the curve used by the node.
    */
   readonly curve: SupportedCurve;
+
+  /**
+   * The specification used to derive this node. Defaults to `bip32` when the
+   * `secp256k1` curve is used, and `slip10` when the `ed25519` curve is used.
+   *
+   * While SLIP-10 and BIP-32 are largely compatible when the `secp256k1` curve
+   * is used, they differ in the way they handle key derivation errors. The
+   * probability of this happening is extremely low, but it is possible. When
+   * full compatibility with one of these specifications is required, this field
+   * should be set to the appropriate value.
+   *
+   * Note that `ed25519` is not supported by BIP-32, so the `bip32`
+   * specification is not available for this curve.
+   */
+  readonly specification: Specification;
```

Comment:
As alternative to the `specification` field, we could change the derivation paths to accept `slip10:${index}`, but that requires some refactoring.